### PR TITLE
Move `_normalize_axis_index` to `cupy/core/internal.pyx`

### PR DESCRIPTION
**Pull request by asi1024**
Discussed on https://github.com/cupy/cupy/pull/4008#issuecomment-694997641.

### CODE REVIEW

**Review by grlee77 — APPROVED** [review]
Looks good. Thanks @asi1024

**Comment by leofang** [comment]
Perhaps we could cythonize the two moved functions, making them cpdef'd?

**Comment by asi1024** [comment]
Made them cpdef.

**Comment by takagi** [comment]
LGTM! Would you resolve the conflict? Then, I'll kick the CI.

**Comment by asi1024** [comment]
Resolved the conflict.

File: cupy/core/_fusion_trace.pyx [hunk 1]
```diff
@@ -13,7 +13,7 @@ from cupy.core._fusion_variable import _TraceArray
 from cupy.core._fusion_variable import _VariableSet
 from cupy.core import _fusion_op
 from cupy.core import _fusion_optimization
-from cupy import _util
+from cupy.core import internal
```
Comment (leofang) [inline cupy/core/_fusion_trace.pyx:16]:
```suggestion
from cupy.core cimport internal
```

File: cupy/core/_routines_math.pyx [hunk 2]
```diff
@@ -7,6 +7,7 @@ from cupy.core._reduction import create_reduction_func
 from cupy.core._kernel import create_ufunc
 from cupy.core._scalar import get_typename
 from cupy.core._ufuncs import elementwise_copy
+from cupy.core import internal
```
Comment (leofang) [inline cupy/core/_routines_math.pyx:10]:
```suggestion
from cupy.core cimport internal
```

**Review by leofang — CHANGES_REQUESTED** [review]

**Comment by leofang** [comment]
Jenkins, test this please

**Review by leofang — APPROVED** [review]
LGTM!

**Pull request merged**